x-pack/plugins/data_visualizer/public/application/common/components/combined_fields/semantic_text.tsx

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState, useEffect, useMemo } from 'react';
+import type { FC } from 'react';
+import type {
+  FindFileStructureResponse,
+  IngestPipeline,
+} from '@kbn/file-upload-plugin/common/types';
+import type { EuiSelectOption } from '@elastic/eui';
+import {
+  EuiButton,
+  EuiFormRow,
+  EuiSelect,
+  EuiSpacer,
+  EuiTextAlign,
+  EuiFieldText,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { cloneDeep } from 'lodash';
+import useDebounce from 'react-use/lib/useDebounce';
+import type {
+  InferenceModelConfigContainer,
+  MappingTypeMapping,
+} from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { createSemanticTextCombinedField, getFieldNames, getNameCollisionMsg } from './utils';
+import { useDataVisualizerKibana } from '../../../kibana_context';
+import type { AddCombinedField } from './combined_fields_form';
+
+interface Props {
+  addCombinedField: AddCombinedField;
+  hasNameCollision: (name: string) => boolean;
+  results: FindFileStructureResponse;
+}
+export const SemanticTextForm: FC<Props> = ({ addCombinedField, hasNameCollision, results }) => {
+  const {
+    services: { http },
+  } = useDataVisualizerKibana();
+  const [inferenceServices, setInferenceServices] = useState<EuiSelectOption[]>([]);
+  const [selectedInference, setSelectedInference] = useState<string | undefined>();
+  const [selectedFieldOption, setSelectedFieldOption] = useState<string | undefined>();
+  const [renameToFieldOption, setRenameToFieldOption] = useState<string>('');
+  const [fieldError, setFieldError] = useState<string | undefined>();
+
+  const fieldOptions = useMemo(
+    () =>
+      getFieldNames(results).map((columnName: string) => {
+        return { value: columnName, text: columnName };
+      }),
+    [results]
+  );
+
+  useEffect(() => {
+    setSelectedFieldOption(fieldOptions[0].value ?? null);
+  }, [fieldOptions]);
+
+  useEffect(() => {
+    http
+      .fetch<InferenceModelConfigContainer[]>('/internal/data_visualizer/inference_services', {
+        method: 'GET',
+        version: '1',
+      })
+      .then((response) => {
+        setInferenceServices(
+          response.map((service) => ({ value: service.model_id, text: service.model_id }))
+        );
+      });
+  }, [http]);
+
+  useEffect(() => {
+    setSelectedInference(String(inferenceServices[0]?.value) ?? undefined);
```

Comment:
Couldn't this setting of the default selected inference happen in the above `useEffect` from the response?
I might be missing some context but looks like this `useEffect` will only be fired once since it depends on `inferenceServices` which only get set in the above useEffect - so this separate `useEffect` may not be needed.

Something like:
```
const inferenceServiceOptions = response.map((service) => ({ value: service.model_id, text: service.model_id }));
setInferenceServices(inferenceServiceOptions);
setSelectedInference(String(inferenceServiceOptions[0]?.value) ?? undefined);
```

Reply:
Updated in https://github.com/elastic/kibana/pull/186956/commits/1b92419b252a87164db31f8a7dcc8bd6ca8211d9